dbsc/workflow/rules/mrtpipelines.smk

```diff
@@ -1,571 +1,550 @@
+        wm_fod=rules.dwi2fod.output.wm_fod,
+        gm_fod=rules.dwi2fod.output.gm_fod,
+        csf_fod=rules.dwi2fod.output.csf_fod,
+        mask=rules.nii2mif.output.mask,
     output:
-        wm_fod=bids(
-            root=join(config['output_dir'], 'mrtpipelines'),
-            datatype='fod',
-            model='csd',
-            desc='wm',
-            suffix='fodnorm.mif',
-            **config['subj_wildcards'],
-        ),
-        gm_fod=bids(
-            root=join(config['output_dir'], 'mrtpipelines'),
-            datatype='fod',
-            model='csd',
-            desc='gm',
-            suffix='fodnorm.mif',
-            **config['subj_wildcards'],
-        ),
-        csf_fod=bids(
-            root=join(config['output_dir'], 'mrtpipelines'),
-            datatype='fod',
-            model='csd',
-            desc='csf',
-            suffix='fodnorm.mif',
-            **config['subj_wildcards'],
-        ),
-    group: "subject_2"
+        wm_fod=expand(mtnormalise_out, tissue="wm", allow_missing=True),
+        gm_fod=expand(mtnormalise_out, tissue="gm", allow_missing=True),
+        csf_fod=expand(mtnormalise_out, tissue="csf", allow_missing=True),
+    threads: workflow.cores
+    group:
+        "subject_2"
     container:
-        config['singularity']['mrtpipelines']
+        config["singularity"]["mrtrix"]
     shell:
-        'mtnormalise -nthreads {params.threads} -mask {input.mask} {input.wm_fod} {output.wm_fod} {input.gm_fod} {output.gm_fod} {input.csf_fod} {output.csf_fod}'
+        "mtnormalise -nthreads {threads} -mask {input.mask} {input.wm_fod} {output.wm_fod} {input.gm_fod} {output.gm_fod} {input.csf_fod} {output.csf_fod}"
+
 
 # DTI (Tensor) Processing
-rule dwi_normalise:
+rule dwinormalise:
     input:
-        dwi=rules.nii_to_mif.output.dwi,
-        mask=rules.nii_to_mif.output.mask,
-    params:
-        threads=workflow.threads,
+        dwi=rules.nii2mif.output.dwi,
+        mask=rules.nii2mif.output.mask,
     output:
         dwi=bids(
-            root=join(config['out_dir'], 'mrtpipelines'),
-            datatype='dwi',
-            space='T1w',
-            desc='norm',
-            suffix='dwi.mif',
-            **config['subj_wildcards'],
-        )
+            root=mrtrix_dir,
+            datatype="dwi",
+            desc="normalized",
+            suffix="dwi.mif",
+            **config["subj_wildcards"],
+        ),
+    threads: workflow.cores
     container:
-        config['singularity']['mrtpipelines']
+        config["singularity"]["mrtrix"]
     shell:
-        'dwinormalise -nthreads {params.threads} {input.dwi} {input.mask} {output.dwi}'
-    group: participant1
+        "dwinormalise individual -nthreads {threads} {input.dwi} {input.mask} {output.dwi}"
+    group:
+        "subject_1"
+
 
+dwi2tensor_out = bids(
+    root=mrtrix_dir,
+    datatype="dti",
+    model="dti",
+    suffix="{dti}.mif",
+    **config["subj_wildcards"],
+)
 
-rule compute_tensor:
+
+rule dwi2tensor:
```

Comment:
This is another rule that could be split into two.

Reply:
Yes agreed, although again, not too much benefit and ultimately this could be an optional rule down the line (may not want tensor measures).